sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java

```diff
@@ -436,6 +445,31 @@ private void clearKubeConfig() {
         this.formatUserKubeConfigsMap.clear();
     }
 
+    @Override
+    public KubernetesClusterImpl withFreeSku() {
+        if (!this.isInCreateMode() && ManagedClusterSkuTier.PREMIUM.equals(this.innerModel().sku().tier())) {
+            this.innerModel().withSupportPlan(KubernetesSupportPlan.KUBERNETES_OFFICIAL);
+        }
```

Comment:
According to the docs, FREE and STANDARD can set the plan to LTS, but not charged until upgrade to PREMIUM:
https://learn.microsoft.com/en-us/azure/aks/long-term-support#enable-long-term-support

If that's true, we probably need a separate API for plan.